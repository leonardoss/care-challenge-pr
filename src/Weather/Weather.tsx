import React, { useState, useEffect } from 'react';
import { Button, Typography, Box } from '@material-ui/core';
import { styled } from '@mui/styles';

const MyBox = styled(Box)({
  background: 'aliceblue',
  padding: '12px',
});

const convert = (temp: any) => {
  let converted = 0;

  if (temp) {
    converted = temp - 273.15
  }

  return converted.toFixed(1);
};

const Weather = (props: any) => {
  const [data, setData] = useState<{ name: string, main: any }>({ name: '', main: {} });
  const [loading, setLoading] = useState(true);
  const [updateTemp, setUpdateTemp] = useState(false);
  const city = props.name

  useEffect(() => {
    fetch('http://api.openweathermap.org/data/2.5/weather?appid=52c005afb94bec44bbeee6101b54ffa4&q=' + city).then((response) => {
      return response.json()
    }).then((data) => {
      setData(data);
      return data
    });

    setLoading(false);
    setUpdateTemp(false)

  }, [city, updateTemp]);

  console.log('##### render Weather');

  return (
    <div style={{
      padding: '2px',
    }}>
      {
        loading ? (
          <p> {props.isLoadingText}</p >
        ) : (
          <MyBox>
            <Typography variant="h4">{data?.name}</Typography>

            <Typography>
              min: {convert(data.main.temp_min)} <sup>°c</sup> | max:{' '}
              {convert(data.main.temp_max)} <sup>°c</sup>
            </Typography>
            <br />
            <Button variant="contained">update temperature</Button>
          </MyBox>
        )}
    </div >
  );
};

export default Weather;