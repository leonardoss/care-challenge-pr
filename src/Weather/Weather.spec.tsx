import React from 'react';
import { render, screen } from '@testing-library/react';
import Weather from './Weather';

test('renders hellow world text', () => {
    const { container } = render(<Weather />);
    expect(container).toMatchSnapshot();
});
