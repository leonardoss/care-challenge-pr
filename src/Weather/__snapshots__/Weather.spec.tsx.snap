// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders hellow world text 1`] = `
<div>
  <div
    style="padding: 2px;"
  >
    <div
      class="ForwardRef(Box)-root-1 MuiBox-root css-0"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      />
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        min: 
        0.0
         
        <sup>
          °c
        </sup>
         | max:
         
        0.0
         
        <sup>
          °c
        </sup>
      </p>
      <br />
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        update temperature
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
